src/PathautoGenerator.php

```diff
@@ -257,19 +257,20 @@ protected function getPatternByEntityType($entity_type_id) {
    * {@inheritdoc}
    */
   public function getPatternByEntity(EntityInterface $entity) {
-    if (!isset($this->patterns[$entity->getEntityTypeId()][$entity->id()])) {
+    $langcode = $entity->language()->getId();
+    if (!isset($this->patterns[$entity->getEntityTypeId()][$entity->id()][$langcode])) {
       foreach ($this->getPatternByEntityType($entity->getEntityTypeId()) as $pattern) {
         if ($pattern->applies($entity)) {
-          $this->patterns[$entity->getEntityTypeId()][$entity->id()] = $pattern;
+          $this->patterns[$entity->getEntityTypeId()][$entity->id()][$langcode] = $pattern;
```

Comment:
Interesting. Wondering if there are other conditions that might result in different patterns for the same entity type. We'll figure out I guess.